File: sprint8/By lessons/lesson4code2/lesson4code2.py
def find_all(text, pattern):
    occurrences = []
    start = 0 # Начнём поиск с начала строки.
    # Найдём первое вхождение, если оно есть.
    while True:
        pos = find(text, pattern, start)
        if pos == -1:
            break
        occurrences.append(pos) # Сохраним вхождение в список.
        start = pos + 1       # И продолжим поиск, начиная с позиции, 
                              # следующей за только что найденной.
    return occurrences

File: sprint8/By lessons/lesson4code2/lesson4code2.kt
fun findAll(text: String, pattern: String): List<Int> {
    val occurrences = mutableListOf<Int>()
    var start = 0 // Начнём поиск с начала строки.
    // Найдём первое вхождение, если оно есть.
    while (true) {
        val pos = find(text, pattern, start)
        if (pos == -1) {
            break
        }
        // Сохраним вхождение в список.
        occurrences.add(pos)
        // И продолжим поиск, начиная с позиции, следующей за только что найденной.
        start = pos + 1
    }
    return occurrences
}
